### DISCUSSION TITLE
Code-Server Blank

### DISCUSSION THREAD

**Discussion by LYF123123** [coder/code-server#4137] [Q&A]
Dear developer, I use self-signed certificates code server 3.11.1 on Centos8 Version 3.32.2, It can work when I use Firefox (78.13.Oser 64bit)on my linux laptap and Chrome(93.0.4577.63) on my Win10 laptap; But I can not use it on my ipad Pro Gen 4th[ipadOS 14.7.1](Both Chrome and Safari). After I enter ther url and the password(It show the password input box), it becomes blank and without anything, I have already tried cmd+R but it is useless, However It can work on my mobile Phone[Honor 20 HamoryOS 2.0 Chrome],Please help me .

## OS/Web Information

- Web Browser

https://user-images.githubusercontent.com/33317812/132814495-7db549cb-adcc-4772-bc2f-6905339025cd.MP4

https://user-images.githubusercontent.com/33317812/132814510-d04d939f-f494-4f43-8951-0b394e7026c1.MP4

: Chrome Safari
- Local OS: IPadOS 14.7.1
- Remote OS: CentOS 8 Version 3.32.2
- Remote Architecture: amd64
- `code-server --version`: 3.11.1

**Comment by jsjoeio** [comment]
Part of me thinks this is because you're using self-signed certificates and they may not be considered secure on iPad.

Have you tried [this](https://support.apple.com/en-us/HT204477)?

I don't know if it's possible to [get the logs](https://imazing.com/guides/how-to-access-iphone-ipad-console-log) from the browser (or the server) but that would help too.

**Reply by LYF123123** [reply]
Dear Developer, I have already add the self-signed certificates in my iPad, but it does not work and still show me blank 
![Uploading 8880B8A6-2151-4251-B433-0EF4C1C8224B.png…]()

**Reply by jsjoeio** [reply]
I hear ya! 

> Have you tried this?

Did you try [this](https://support.apple.com/en-us/HT204477)?

**Reply by LYF123123** [reply]
![4AD5E29D-B353-4EC6-B310-B38F3C98D0DF](https://user-images.githubusercontent.com/33317812/132915588-f8b66b4c-d798-4799-a3f2-9959b5f5b109.png)
Dear Developer, I  have already enable full trust for root certificates, but it still show me blank in Safari and Chrome

**Reply by jsjoeio** [reply]
Great! Next, can you try [accessing the browser logs](https://imazing.com/guides/how-to-access-iphone-ipad-console-log) in Safari and Chrome to see if that tells us anything?

**Reply by LYF123123** [reply]
Dear Developer , I find I cannot use self-signed cert on my linux server, I cannot set local dns to access my server because my server is in the LAN, and I try the PWA on AppStore, However it is still useless.😭  Maybe I should stop use the SSL? Is there any risk without SSL? I really do not understand how to set self-signed certs. Can you help me 😭

**Reply by jsjoeio** [reply]
Ahhh... I wish I could help more, but self-signed certs, SSL, etc is a bit out of my expertise 😢 @code-asher might be able to offer suggestions though.

Alternatively, you can check out our [deploy-code-server](https://github.com/cdr/deploy-code-server) repo for other deployment options, which might avoid this problem entirely.

See [this video](https://www.youtube.com/watch?v=IWfkkaY5fXs) for setting up Railway to use with code-server.

**Reply by code-asher** [reply]
I would recommend using a real domain and getting a certificate with Let's Encrypt, one of the alternatives @jsjoeio mentioned, or using `code-server --link` to handle the TLS for you automatically.

Self-signed certificates seem to be tricky especially with iOS.
